sqlx-sqlite/src/index.rs

```diff
@@ -0,0 +1,599 @@
+                        let max_values = max_values.as_primitive::<Int64Type>();
+                        let min = min_values.value(row_group);
+                        let max = max_values.value(row_group);
+                        let column_statistics = ColumnStatistics {
+                            null_count: null_counts.value(row_group) as i64,
+                            stats: MinMaxStats::Int(min, max),
+                        };
+                        row_group_statistics[row_group]
+                            .column_statistics
+                            .push(column_statistics);
+                    }
+                    datafusion::arrow::datatypes::DataType::Utf8 => {
+                        let min_values = min_values.as_string::<i32>();
+                        let max_values = max_values.as_string::<i32>();
+                        let min = min_values.value(row_group).to_string();
+                        let max = max_values.value(row_group).to_string();
+                        let column_statistics = ColumnStatistics {
+                            null_count: null_counts.value(row_group) as i64,
+                            stats: MinMaxStats::String(min, max),
+                        };
+                        row_group_statistics[row_group]
+                            .column_statistics
+                            .push(column_statistics);
+                    }
+                    datafusion::arrow::datatypes::DataType::LargeUtf8 => {
+                        let min_values = min_values.as_string::<i64>();
+                        let max_values = max_values.as_string::<i64>();
+                        let min = min_values.value(row_group).to_string();
+                        let max = max_values.value(row_group).to_string();
+                        let column_statistics = ColumnStatistics {
+                            null_count: null_counts.value(row_group) as i64,
+                            stats: MinMaxStats::String(min, max),
+                        };
+                        row_group_statistics[row_group]
+                            .column_statistics
+                            .push(column_statistics);
+                    }
+                    _ => {} // ignore other types, we just don't put them in the index and filters will not be pushed down
+                }
+            }
+        }
+
+        let file_statistics = FileStatisticsInsert {
+            file_name: file_name.to_string(),
+            file_size_bytes: file_size as i64,
+            row_group_count: metadata.num_row_groups() as i64,
+            row_count: metadata.file_metadata().num_rows(),
+        };
+
+        self.add_row(file_statistics, row_group_statistics).await?;
+        Ok(())
+    }
+
+    async fn add_row(
+        &self,
+        file_statistics: FileStatisticsInsert,
+        row_group_statistics: Vec<RowGroupStatisticsInsert>,
+    ) -> anyhow::Result<()> {
+        self.initialize().await?;
```

Comment:
Should this also be called when pruning, just in case no data exists yet?

Reply:
Let me throw it into `new()` then.

Reply:
d47bbce